languages/python/oso/polar/polar.py

```diff
@@ -109,6 +109,7 @@ def join(separator, left, right):
                 query = self.ffi_polar.next_inline_query()
                 if query is None:  # Load is done
                     break
+                query.set_message_enricher(self.host.enrich_message)
```

Comment:
You'll need to use the copied host made on L114 here. If the inline query creates external instances they are attached to that one instead of `self.host`.